src/ethernet_interface.cpp

```diff
@@ -133,6 +189,43 @@ EthernetInterface::EthernetInterface(stdplus::PinnedRef<sdbusplus::bus_t> bus,
     {
         addStaticRoute(staticRoute);
     }
+
+    const std::filesystem::path dir = "/sys/class/net";
+    for (auto&& d : std::filesystem::directory_iterator(dir))
+    {
+        std::filesystem::path ifindex = d.path() / "ifindex";
+        int fd = open(ifindex.c_str(), O_RDONLY);
```

Comment:
Since I already have the open file descriptor from the event loop, I think using the syscalls makes more sense, especially since I need to both read and write. What do you think?